src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -1225,16 +1225,9 @@ namespace QgsWfs
           QDomElement envElem = QgsOgcUtils::rectangleToGMLEnvelope( rect, doc, srsName, invertAxis, prec );
           if ( !envElem.isNull() )
           {
-            if ( crs.isValid() )
+            if ( crs.isValid() && srsName.isEmpty() )
```

Comment:
why removing "envElem.setAttribute( QStringLiteral( "srsName" ), srsName );" when crs.isValid() and !srsName.isEmpty() ?

Reply:
Because it is already done in `QgsOgcUtils::rectangleToGMLEnvelope`, the `srsName` attribute is already set to `params.srsName` if not empty